helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -200,14 +196,12 @@ public void computeScore(Map<String, Integer> overallClusterRemainingCapMap) {
       // score = SUM(weight * (resource_capacity/cluster_capacity) where weight = 1/(1-total_util%)
       // it could be be simplified to "resource_capacity/cluster_remainingCapacity".
       for (Map.Entry<String, Integer> resourceCapacity : _replica.getCapacity().entrySet()) {
+        String capacityKey = resourceCapacity.getKey();
         if (resourceCapacity.getValue() == 0) {
           continue;
         }
-        score = (overallClusterRemainingCapMap.get(resourceCapacity.getKey()) == 0
-            || resourceCapacity.getValue() > (overallClusterRemainingCapMap
-            .get(resourceCapacity.getKey()))) ? Float.MAX_VALUE
-            : score + (float) resourceCapacity.getValue() / (overallClusterRemainingCapMap
-                .get(resourceCapacity.getKey()));
+        score = score + (float) resourceCapacity.getValue() / (overallClusterRemainingCapMap
```

Comment:
I realized that you still need to handle the 0 case here, and it's actually not straightforward. 

If the remaining is 0 for a capacity key, what should be the score here? We can't do max_value since we must also differentiate between different weights.

Reply:
Correct..I missed that..